api/server/server.go

```diff
@@ -935,9 +935,8 @@ func (s *Server) postContainersCreate(version version.Version, w http.ResponseWr
 	if err != nil {
 		return err
 	}
-	adjustCpuShares(version, hostConfig)
 
-	containerId, warnings, err := s.daemon.ContainerCreate(name, config, hostConfig)
+	containerId, warnings, err := s.daemon.ContainerCreate(name, config, hostConfig, version)
```

Comment:
I'd rather have a boolean passed in from the controller  which drives the cpu share adjustment instead of leaking the api version to the daemon